measurements.py

```diff
@@ -17,11 +19,26 @@ def is_clipped(data: AnalogSpan):
         or data.max == data._adc_to_voltage(data.acquisition_info.max_adc_code)
 
 
+def mean(data: AnalogSpan):
+    total_sum = 0
+    total_samples = 0
+
+    for chunk in data.raw_chunks():
+        total_sum += chunk._convert_sample(np.sum(chunk.raw_samples, dtype=np.int64)) + chunk.voltage_transform.offset * len(chunk.raw_samples)
+        total_samples += len(chunk.raw_samples)
+
+    return total_sum / total_samples
+
+
 def rms(data: AnalogSpan):
-    # TODO: numpy accelerate
-    sum_of_squares = sum(x**2 for x in data)
-    rms = math.sqrt(sum_of_squares / len(data))
-    return rms
+    total_sum = 0
+    total_samples = 0
+
+    for chunk in data.raw_chunks():
+        total_sum += np.sum(chunk.samples ** 2)
```

Comment:
hm...

RMS = (1/N*(sum(Y...^2))^0.5

Seems like you will get a different result using the ADC code instead of the voltage.

sum(Y...^2)
sum((m*x + b)^2)
sum(x^2*m^2+2*b*m*x+b^2)

sum(b^2) = N*b^2
sum(2bmx)=2bm*sum(x)
sum(m^2*x^2)=m^2*sum(x^2)

I think that becomes:
RMS = `( 1/N * ((N*b^2)+(2bm*sum(x))+(m^2*sum(x^2))))^0.5`

does that sound right?

Does that look right?

Reply:
`chunk.samples` is in volts. We should benchmark `( 1/N * ((N*b^2)+(2bm*sum(x))+(m^2*sum(x^2))))^0.5` against this approach, though. Avoiding the voltage conversion could be a net win.